### ISSUE TITLE
cannot stream video or audio content

### ISSUE THREAD

**Issue by MacBiter** [i3roly/firefox-dynasty#12]
The following sites all fail to stream content : 

Amazon Prime 
BBC iPlayer
Liverpool FC
ITVX
Channel 4 
Audible 

YouTube works, but that's the only streaming site I use that does.

**Comment by MacBiter** [comment]
I've gone back to Firefox 78 ESR - even though I can't stream DRM content there, I can at least stream BBC iPlayer, Liverpool FCTV, and my Preferences / Security / saved passwords are both visible and copiable.

I'd be happy to use Firefox Dynasty again if these issues were resolved. :-)

**Comment by i3roly** [comment]
i have no idea how to test anything you're saying doesn't work, as they all require logins and are region locked.

i did, however, try the bbc iplayer radio and it seems fine. i can hear it and everything.


https://github.com/user-attachments/assets/b36ded04-f9bd-4108-9dbc-d2bbbe9be25f

**Comment by Wowfunhappy** [comment]
Before the whole browser freezes (see #13), videos on the homepage of nytimes.com also show a playback error in Firefox Dynasty.

I noticed this message in Terminal which seems relevent:

> [Child 13188, MediaDecoderStateMachine #1] WARNING: Decoder=121cc0c00 Decode error: NS_ERROR_DOM_MEDIA_FATAL_ERR (0x806e0005) - MediaResult mozilla::AppleVTDecoder::InitializeSession(): Couldn't create decompression session!: file /Users/Gagan/Downloads/mozilla-unified/dom/media/MediaDecoderStateMachineBase.cpp:167

**Comment by i3roly** [comment]
you should turn on media logging in about:logging and try to find more.

i did encounter the same issue on nytimes articles as you're mentioning on the 10.8 vm, but the error does not seem to be about the actual media codecs. i was able to download and play the video without issue, it has to do with some kind of autoplay feature in html5 that the system is having trouble understanding

```
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac4800 Pause() called by JS
[Child 1474: Main Thread]: D/HTMLMediaElement Loading due to Pause()
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac4800 Network state changed to NO_SOURCE
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac4800 Network state changed to EMPTY
[Child 1474: MediaDecoderStateMachine #2]: D/AudioSinkWrapper 12bdad300: AudioSinkWrapper::OnMuted(false)
[Child 1474: MediaDecoderStateMachine #2]: D/AudioSinkWrapper 12bdad300: AudioSinkWrapper::OnMuted, but no audio track
[Child 1474: Main Thread]: D/HTMLMediaElementEvents 124ac3800 Dispatching event volumechange
  JavaScript warning: https://www.googletagmanager.com/gtm.js?id=GTM-P528B3&gtm_auth=tfAzqo1rYDLgYhmTnSjPqw&gtm_preview=env-130&gtm_cookies_win=x, line 1: unreachable code after return statement
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac4800 SuspendOrResumeElement(suspend=0) docHidden=1
[Child 1474: Main Thread]: D/MediaDecoder MediaDecoder[127c2d400] UpdateVideoDecodeMode(), set Suspend because of invisible element.
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac3800 SuspendOrResumeElement(suspend=0) docHidden=1
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac4800 SuspendOrResumeElement(suspend=0) docHidden=1
[Child 1474: Main Thread]: D/MediaDecoder MediaDecoder[127c2d400] UpdateVideoDecodeMode(), set Suspend because of invisible element.
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac3800 SuspendOrResumeElement(suspend=0) docHidden=1
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac4800 SuspendOrResumeElement(suspend=1) docHidden=1
[Child 1474: Main Thread]: D/HTMLMediaElementEvents 124ac4800 unblock event delivery
[Child 1474: Main Thread]: D/MediaDecoder MediaDecoder[127c2d400] UpdateVideoDecodeMode(), set Suspend because of invisible element.
[Child 1474: Main Thread]: D/HTMLMediaElement 124ac3800 SuspendOrResumeElement(suspend=1) docHidden=1
[Child 1474: Main Thread]: D/MediaDecoder MediaDecoder[127c2d400] Pause
[Child 1474: Main Thread]: D/MediaDecoder MediaDecoder[127c2d400] SetDelaySeekMode, shouldDelaySeek=1
[Child 1474: Main Thread]: D/HTMLMediaElementEvents 124ac3800 unblock event delivery
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), VideoDecodeMode=(Normal->Suspend), mVideoDecodeSuspended=F
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), VideoDecodeMode=(Suspend->Suspend), mVideoDecodeSuspended=F
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), early return because the mode does not change
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), VideoDecodeMode=(Suspend->Suspend), mVideoDecodeSuspended=F
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), early return because the mode does not change
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SuspendMediaSink
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder VideoSink=12758f2e0 [Shutdown]
[Child 1474: Main Thread]: D/HTMLMediaElementEvents 124ac3800 Queuing event mozstartvideosuspendtimer
[Child 1474: Main Thread]: D/HTMLMediaElementEvents 124ac3800 postpone runner nsAsyncEventRunner for mozstartvideosuspendtimer
[Child 1474: Main Thread]: D/HTMLMediaElement OnVisibilityChange(): ApproximatelyNonVisible
[Child 1474: Main Thread]: D/MediaDecoder MediaDecoder[127c2d400] UpdateVideoDecodeMode(), set Suspend because of invisible element.
[Child 1474: Main Thread]: D/HTMLMediaElement OnVisibilityChange(): ApproximatelyNonVisible
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), VideoDecodeMode=(Suspend->Suspend), mVideoDecodeSuspended=F
[Child 1474: MediaDecoderStateMachine #3]: D/MediaDecoder MediaDecoderStateMachine[127c9d000] Decoder=127c2d400 SetVideoDecodeModeInternal(), early return because the mode does not change
```

**Comment by MacBiter** [comment]
The other issue (possibly unconnected?) is that when I go to Preferences / Saved logins and passwords, the eye button to show the password doesn't work, nor does the Copy button next to it.

I should point out that I didn't install Dynasty as a new 'empty page' app but either it took over my 78 ESR Profile, or else I imported it (can't now remember) - I do know that I ended up with all my ESR windows, tabs, bookmarks, etc. I don't know if this is relevant?

**Comment by i3roly** [comment]
it may be. you should try to create a new user profile and see if those passwords show up.

it wouldn’t surprise me if the encryption approach changed in ~50 releases.

everything else should import, but the problem may be that the newer releases changed how they hash passwords for “more security” and that would make your old saved passwords undecipherable to the new firefox.
	-i haven’t checked this theory though. i just think it would be unsurprising if that’s the case.

Thanks,
Gagan

**Comment by Wowfunhappy** [comment]
> The other issue (possibly unconnected?) is that when I go to Preferences / Saved logins and passwords, the eye button to show the password doesn't work, nor does the Copy button next to it.

This would almost certainly be a separate issue, but fwiw I can't replicate it, the button works fine for me.

Also (now even more off topic, I'm terrible at this) I really do recommend migrating a cross-browser password manager such as Bitwarden, it means that long term you won't be tied to any one platform.

**Comment by MacBiter** [comment]
Just to check first - Dynasty won't try and take over my 78 ESR Profile if I run it immediately after 78? In other words I can create a new profile and go back to 78 if I need? (I've got a feeling the Profiles are separate anyway in Application Support).

**Comment by i3roly** [comment]
no i mean create a new user account to avoid that possibility.

if you want to ensure it won’t clobber your profile, go to Contents/MacOS and run ./firefox —ProfileManager where it will let you create and use a new one, but you’ll probably have to run it again to delete it and set your 78 ESR as default afterwards.

it does sound like a hashing algorithm change between 78 and the most recent stable release, though.

Thanks,
Gagan

**Comment by MacBiter** [comment]
ok, I have other user accounts already, I'll check there - tomorrow or the weekend, it's my bedtime!

**Comment by MacBiter** [comment]
bad news I'm afraid.

none of the sites mentioned play content, e.g. Liverpool FC which still works in 78 ESR. BBC iPlayer radio DOES play as you found, but I'm not sure I tested that before?,  in other words it might have played in my main account.

and the Copy /Eyeball password buttons still don't work.

this was using a clean version of Dynasty in another user account.

**Comment by i3roly** [comment]
The video thing I'm aware of. It's probably tied to the same thing wowfunhappy said about the nytimes not working. I'll look closer at the diffs for vtdecoder between 78 and now. With respect to the passwords thing, I'm almost positive it's something on your end. They work for me and another person.                                                                                                                                                                                                                                                                                                                                         Sent from my BlackBerry 10 smartphone on the Rogers network.                                                                                                                                                                                                                From: MacBiterSent: Friday, 4 October 2024 9:20 AMTo: i3roly/firefox-dynastyReply To: i3roly/firefox-dynastyCc: gagan sidhu; CommentSubject: Re: [i3roly/firefox-dynasty] cannot stream video or audio content (Issue #12)
bad news I'm afraid.
none of the sites mentioned play content, e.g. Liverpool FC which still works in 78 ESR. BBC iPlayer radio DOES play as you found, but I'm not sure I tested that before?,  in other words it might have played in my main account.
and the Copy /Eyeball password buttons still don't work.
this was using a clean version of Dynasty in another user account.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/i3roly/firefox-dynasty/issues/12#issuecomment-2393947684",
"url": "https://github.com/i3roly/firefox-dynasty/issues/12#issuecomment-2393947684",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by MacBiter** [comment]
"With respect to the passwords thing, I'm almost positive it's something on your end. They work for me and another person."

I'm puzzled what it could be as they work fine in 78 ESR in my main account?

**Comment by i3roly** [comment]
i told you already lol. the chances are that the HASHING of the passwords from 78 CHANGED at some point afterwards.

this would render your OLD SAVED passwords USELESS in terms of expecting to see their decrypted value and/or possibly even using them ON A NEW FIREFOX VERSION.

in other words, the way your passwords were encrypted and stored (using hashes) likely changed after 78, which means that these saved passwords would not show up under the eyeball because the mapping has changed



Thanks,
Gagan

**Comment by Wowfunhappy** [comment]
I think MacBiter is saying the eyeball still doesn't work even in a fresh user account, which would imply no saved passwords from 78. This is assuming they aren't syncing a Mozilla account or something.

Which is very strange because as I said, it works for me!

**Comment by MacBiter** [comment]
That user account has rarely been used but I'd better check I didn't use 78 ESR once or twice in the past (I have 78 in Applications as Firefox, and Dynasty as Firefox 131) - I'll check tomorrow.

**Comment by MacBiter** [comment]
To be sure, I created a new user and logged into a couple of sites. I then quit Dynasty and relaunched - the situation is exactly the same: when I click the eyeball next to Password it doesn't show and the eyeball gets a diagonal line through it. If I then click Copy, nothing happens. But if I click Copy next to Username, that works fine.

Yet in 78ESR, non-DRM video streams, and the Settings / Saved passwords works.

**Comment by i3roly** [comment]
you have two people saying it works, and you are the only one reporting it not working.

i've tried it on all of my VMs without issue. this is something to do with your existing firefox settings clashing with the new firefox. i cannot say this enough times. it is nothing to do with this build

you cannot equate the video issues as equivalent to the password saving issues, because i didn't touch anything related to the latter and extremely little in the former. the situation for the videos is they are encrypted and firefox has changed their video decoding approach in the four years since 78ESR, which requires some adaptations. 

it is likely these video issues are occurring on 10.12 and lower but i do not care to install a 10.12 vm to check just where the issue starts, since addressing that is important.

in contrast, the passwords issue is a non-issue. i am sure wowfunhappy also tried to duplicate your situation and failed to do so, as i have. firefox reads more than just ~/Library. it reads a lot of things, and it's possible your existing database from 78ESR is somehow causing problems. i can't say for sure, what i can say is it's not the software.

i am sure if you asked mozilla, by posting a ticket saying you've tried to use your 78ESR profile on a newer mac with 130, they would tell you the same thing.

https://github.com/user-attachments/assets/2d77435d-ab06-4325-8322-0ede88302a55

**Comment by i3roly** [comment]
new version will be up in a few minutes with the video issue fixed, also the checkboxes will render properly as will the buttons with the default focusrings.

enjoy you hungry hungry fucking hippos @wowfunhappy @macbiter lol


https://github.com/user-attachments/assets/a99ecb2d-6e6c-4327-bdf6-890dbe230a51
